subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1447,6 +1821,752 @@ static int tls_mbedtls_init(struct tls_context *context, bool is_server)
+                (int)context->options.dtls_handshake_timeout_min) != WOLFSSL_SUCCESS) {
+            return -EINVAL;
+        }
+
+        /* Underlying socket always acts like non-blocking due to IO callback overrides */
+        wolfSSL_dtls_set_using_nonblock(context->wssl, 1);
+    }
+
+    return 0;
+}
+#endif
+
+static int tls_wolfssl_set_options(struct tls_context *context)
+{
+    int ret = 0;
+    bool is_server = FALSE;
+
+    if (context->options.role == ZTLS_IS_SERVER) {
+        is_server = TRUE;
+    }
+
+    ret = tls_wolfssl_set_verify(context);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = tls_wolfssl_set_hostname(context);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = tls_wolfssl_set_session_cache_mode(context);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = tls_wolfssl_set_ciphersuites(context);
+    if (ret != 0) {
+        return ret;
+    }
+
+#ifdef HAVE_ALPN
+    ret = tls_wolfssl_set_alpn(context);
+    if (ret != 0) {
+        return ret;
+    }
+#endif
+
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+    ret = tls_wolfssl_set_dtls_timeouts(context);
+    if (ret != 0) {
+        return ret;
+    }
+#endif
+
+    return 0;
+}
+
+static int tls_wolfssl_init(struct tls_context *context, bool is_server)
+{
+    int ret;
+
+    context->options.role = is_server ? ZTLS_IS_SERVER : ZTLS_IS_CLIENT;
+
+#if defined(CONFIG_WOLFSSL_DEBUG)
+    wolfSSL_Debugging_ON();
+#endif
+
+    if (context->type == SOCK_STREAM) {
+        if (is_server) {
+            if (NULL == context->ctx)
+                context->ctx = wolfSSL_CTX_new(wolfTLSv1_2_server_method());
+        }
+        else {
+            if (NULL == context->ctx)
+                context->ctx = wolfSSL_CTX_new(wolfTLSv1_2_client_method());
+        }
+
+        wolfSSL_CTX_SetIORecv(context->ctx, tls_wolf_rx);
+        wolfSSL_CTX_SetIOSend(context->ctx, tls_wolf_tx);
+    }
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+    else if (context->type == SOCK_DGRAM) {
+#ifdef WOLFSSL_DTLS
+        if (is_server) {
+            if (NULL == context->ctx)
+                context->ctx = wolfSSL_CTX_new(wolfDTLSv1_2_server_method());
+        }
+        else {
+            if (NULL == context->ctx)
+                context->ctx = wolfSSL_CTX_new(wolfDTLSv1_2_client_method());
+        }
+        wolfSSL_CTX_SetIORecv(context->ctx, dtls_wolf_rx);
+        wolfSSL_CTX_SetIOSend(context->ctx, dtls_wolf_tx);
+#else
+    return -ENOTSUP;
+#endif
+    }
+#endif
+    else {
+        return -EINVAL;
+    }
+
+    ret = tls_wolfssl_set_credentials(context);
+    if (ret != 0) {
+        return -EINVAL;
+    }
+
+#ifndef NO_PSK
+    if (NULL != context->psk) {
+        if (is_server) {
+            wolfSSL_CTX_set_psk_server_callback(context->ctx, tls_psk_server_cb);
+            /* TODO: review psk hint */
+            wolfSSL_CTX_use_psk_identity_hint(context->ctx, "zephyr wolfssl TLS server");
```

Comment:
What does mbeTLS use for their PSK server identity hint? It's fairly specific to the examples I think.

Reply:
Mbedtls seems to set no server identity hint upon closer inspection, I will update the code to do the same.